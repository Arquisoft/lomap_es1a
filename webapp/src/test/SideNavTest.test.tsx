import { render, fireEvent } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import SideNav from '../components/Sidenav';

describe('SideNav', () => {
  it('should render the component', () => {
    const { container } = render(
      <BrowserRouter>
        <SideNav />
      </BrowserRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  /*t('should toggle the open state when the menu button is clicked', () => {
    const { getByRole } = render(
      <BrowserRouter>
        <SideNav />
      </BrowserRouter>
    );
    const menuButton = getByRole('button');
    fireEvent.click(menuButton);
    expect(menuButton.getAttribute('aria-expanded')).toBe('true');
  });
  */
});
