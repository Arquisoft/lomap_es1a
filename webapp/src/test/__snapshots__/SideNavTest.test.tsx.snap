// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNav should render the component 1`] = `
<div
  class="sidenavClosed"
  style="background-color: rgb(25, 118, 210);"
>
  <button
    class="menuBtn"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="KeyboardDoubleArrowRightIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M6.41 6 5 7.41 9.58 12 5 16.59 6.41 18l6-6z"
      />
      <path
        d="m13 6-1.41 1.41L16.17 12l-4.58 4.59L13 18l6-6z"
      />
    </svg>
  </button>
  <a
    aria-current="page"
    class="sideitem active"
    href="/"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="TravelExploreIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19.3 16.9c.4-.7.7-1.5.7-2.4 0-2.5-2-4.5-4.5-4.5S11 12 11 14.5s2 4.5 4.5 4.5c.9 0 1.7-.3 2.4-.7l3.2 3.2 1.4-1.4-3.2-3.2zm-3.8.1c-1.4 0-2.5-1.1-2.5-2.5s1.1-2.5 2.5-2.5 2.5 1.1 2.5 2.5-1.1 2.5-2.5 2.5zM12 20v2C6.48 22 2 17.52 2 12S6.48 2 12 2c4.84 0 8.87 3.44 9.8 8h-2.07c-.64-2.46-2.4-4.47-4.73-5.41V5c0 1.1-.9 2-2 2h-2v2c0 .55-.45 1-1 1H8v2h2v3H9l-4.79-4.79C4.08 10.79 4 11.38 4 12c0 4.41 3.59 8 8 8z"
      />
    </svg>
    <span
      class="linkTextClosed"
    >
      Map
    </span>
  </a>
  <a
    class="sideitem"
    href="/account"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="AccountCircleIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 4c1.93 0 3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5S10.07 6 12 6zm0 14c-2.03 0-4.43-.82-6.14-2.88C7.55 15.8 9.68 15 12 15s4.45.8 6.14 2.12C16.43 19.18 14.03 20 12 20z"
      />
    </svg>
    <span
      class="linkTextClosed"
    >
      Account
    </span>
  </a>
  <a
    class="sideitem"
    href="/friends"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="EmojiPeopleIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <circle
        cx="12"
        cy="4"
        r="2"
      />
      <path
        d="M15.89 8.11C15.5 7.72 14.83 7 13.53 7h-2.54C8.24 6.99 6 4.75 6 2H4c0 3.16 2.11 5.84 5 6.71V22h2v-6h2v6h2V10.05L18.95 14l1.41-1.41-4.47-4.48z"
      />
    </svg>
    <span
      class="linkTextClosed"
    >
      Friends
    </span>
  </a>
  <a
    class="sideitem"
    href="/about"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="InfoIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
      />
    </svg>
    <span
      class="linkTextClosed"
    >
      About
    </span>
  </a>
</div>
`;
